src/Command/CreateSnapshotsCommand.php

```diff
@@ -24,9 +25,18 @@
  * @author Thomas Rabaix <thomas.rabaix@sonata-project.org>
  *
  * @final since sonata-project/page-bundle 3.26
+ * NEXT_MAJOR: Remove code related with mode option!
  */
 class CreateSnapshotsCommand extends BaseCommand
 {
+    private SnapshotManagerInterface $snapshotManager;
+
+    public function __construct(string $name = null, SnapshotManagerInterface $snapshotManager)
+    {
+        parent::__construct($name);
+        $this->snapshotManager = $snapshotManager;
+    }
```

Comment:
```suggestion
    public function __construct(SnapshotManagerInterface $snapshotManager)
    {
        parent::__construct();
        $this->snapshotManager = $snapshotManager;
    }
```
Seems ok to me.


But you'll have to support all the possible call
```
new Command(); // To deprecate
new Command($name); // To deprecate
new Command($snapshotManager);
```

Doing
```
public function __construct($snapshotManagerOrDeprecatedName = null)
    {
        if (snapshotManagerOrDeprecatedName instanceof SnapshotManagerInterface) {}
        elseif (is_string(snapshotManagerOrDeprecatedName) || === null) { // deprecation }
    }
```

Reply:
hmmmm I get it, because `CreateSnapshotsCommand` is not a final class :'( 

I'll add some support for this, thanks for the reminder ;)

Reply:
I changed this ;)